src/lib/hooks/useConnection.native.js

```diff
@@ -1,18 +1,15 @@
 import { useEffect, useState } from 'react'
-import NetInfo from '@react-native-community/netinfo'
+import { addEventListener, fetch } from '@react-native-community/netinfo'
 
 export default () => {
   const [isConnection, setIsConnection] = useState(true)
 
-  NetInfo.fetch().then(({ isConnected }) => {
```

Comment:
removed fetch() call from each render, doing it once inside effect then update status in addEventListener